### PR TITLE
feat(reverseproxy): updated DB amd api's to store the connection status

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
BREAKING CHANGE: 'metadata' api route is removed.

### CODE REVIEW

File: src/db/device.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+          device.connectionStatus
+        ])
+      if (results.rowCount > 0) {
+        return await this.getById(device.guid)
+      }
+      return null
+    } catch (error) {
+      log.error(`Failed to insert: ${device.guid}`, error)
+      throw new Error(error)
+    //   if (error.code === '23505') { // Unique key violation
```
Comment (rsdmike) [inline src/db/device.ts:86]: We prob should still handle this particular error?

File: src/db/device.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          device.hostname,
+          device.mpsInstance,
+          device.connectionStatus
+        ])
+      if (results.rowCount > 0) {
+        return await this.getById(device.guid)
+      }
+      return null
+    } catch (error) {
+      log.error(`Failed to update: ${device.guid}`, error)
```
Comment (rsdmike) [inline src/db/device.ts:113]: Missing return when error

File: src/db/device.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    }
+  }
+
+  /**
+  * @description Clear the mpsInstance for associated devices before process exit
+  * @param {string} mpsInstance
+  * @returns {void}
+  */
+  updateByInstance (mpsInstance: string): void {
```
Comment (rsdmike) [inline src/db/device.ts:122]: I think renaming this to be more accurate would be good. Like clearInstanceStatus

File: src/routes/devices/index.ts [hunk 4]
```diff
@@ -4,17 +4,26 @@
...
+import { param } from 'express-validator'
 
 const deviceRouter: Router = Router()
 
 deviceRouter.get('/', metadataQueryValidator(), getAll)
 deviceRouter.get('/stats', stats)
-deviceRouter.delete('/disconnect/:guid', disconnect)
-deviceRouter.get('/:guid', deviceGetValidator(), get)
+deviceRouter.get('/tags', tags)
+deviceRouter.get('/:guid', param('guid').isUUID(), get)
```
Comment (rsdmike) [inline src/routes/devices/index.ts:23]: Let's call the function getDevice instead of get to match the others

File: src/routes/devices/stats.ts [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
 export async function stats (req, res): Promise<void> {
-  const amtCredentials = await req.mpsService.db.getAllAmtCredentials()
+  const db = new DeviceDb()
+  const devices = await db.get()
   let connectedCount = 0
   let totalCount = 0
-  if (amtCredentials != null) {
-    totalCount = Object.keys(amtCredentials).length
+  if (devices != null) {
+    totalCount = Object.keys(devices).length
```
Comment (rsdmike) [inline src/routes/devices/stats.ts:13]: I don't think this is correct. Devices is an array now right? Should be just .length without need for object keys

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/routes/devices/index.ts (hunk 4 above)
Comment (madhavilosetty-intel) [reply src/routes/devices/index.ts:23]: done

File: src/db/device.ts (hunk 3 above)
Comment (madhavilosetty-intel) [reply src/db/device.ts:122]: done

File: src/routes/devices/stats.ts (hunk 5 above)
Comment (madhavilosetty-intel) [reply src/routes/devices/stats.ts:13]: done

File: src/db/device.ts (hunk 1 above)
Comment (madhavilosetty-intel) [reply src/db/device.ts]: done

File: src/db/device.ts (hunk 2 above)
Comment (madhavilosetty-intel) [reply src/db/device.ts:113]: done

File: src/routes/devices/update.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import { DeviceDb } from '../../db/device'
+import { logger as log } from '../../utils/logger'
+import { MPSValidationError } from '../../utils/MPSValidationError'
+
+export async function updateDevice (req, res): Promise<void> {
+  const db = new DeviceDb()
+  try {
+    const errors = validationResult(req)
+    if (!errors.isEmpty()) {
+      res.status(400).json({ errors: errors.array() })
```
Comment (brianosburn-intel) [inline src/routes/devices/update.ts:15]: should this have a .end()

**Review by brianosburn-intel — APPROVED** [review]

**Review by rsdmike — APPROVED** [review]

**Pull request merged**